🐛 Bug Report: Getting an error when creating a new project

### 📜 Description

Installation with **`npx @backstage/create-app@latest`** failed

### 👍 Expected behavior

It should complete the installation

### 👎 Actual Behavior with Screenshots

When I run the command **`npx @backstage/create-app@latest`**, I get:

```
executing     yarn install ✖
Error: Could not execute command yarn install
```

Then I reran a **`yarn install`** where I got this error:

```
error /Users/abcde/Desktop/backstage/node_modules/@swc/core: Command failed.
Exit code: 1
Command: node postinstall.js
Arguments:
Directory: /Users/abcde/Desktop/backstage/node_modules/@swc/core
Output:
node:internal/modules/cjs/loader:1148
  throw err;
  ^

Error: Cannot find module '/Users/abcde/Desktop/backstage/node_modules/@swc/core/postinstall.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
    at Module._load (node:internal/modules/cjs/loader:986:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:174:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

### 👟 Reproduction steps

1 - `npx @backstage/create-app@latest`
2 - `yarn install`

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

Windows 10 & macOS 14.5 yield the same result

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

Another data point FWIW - in case it helps someone else in the future.

I had problems with different yarn version and node versions.  
yarn version unknown (didnt check)
node version 20.14.0
nvm version 0.39.1 (nvm version shouldnt matter).
the above combination generated similar error's as OP when running `yarn dev` - webpack would not compile with `[0] Error: Cannot find module '@swc/types'`

`yarn upgrade` and the below combination, `yarn dev` was successful and `webpack compiled successfully` and 
yarn 1.22.19
nvm 0.39.1
node v20.13.1

@awanlin this issue still exists. @backstage/create-app@latest will complete successfully but during a yarn build:backend it fails. 
I do not see the missing module in an of the package.json files.

`root@bc2468916414:/app# grep -i 'swc/types' package.json
root@bc2468916414:/app# grep -i 'swc/types' packages/backend/package.json
root@bc2468916414:/app# grep -i 'swc/types' packages/app/package.json`

`8.486 app:  Error: Failed to compile './src/index.tsx':
8.486 app:    Error: Cannot find module '@swc/types'
8.486 app:  Require stack:
8.486 app:  - /app/node_modules/@swc/core/index.js
8.486 app:  - /app/node_modules/swc-loader/src/index.js
8.486 app:  - /app/node_modules/loader-runner/lib/loadLoader.js
8.486 app:  - /app/node_modules/loader-runner/lib/LoaderRunner.js
8.486 app:  - /app/node_modules/webpack/lib/NormalModule.js
8.486 app:  - /app/node_modules/webpack/lib/ModuleFilenameHelpers.js
8.486 app:  - /app/node_modules/esbuild-loader/dist/index.cjs
8.486 app:  - /app/node_modules/@backstage/cli/dist/cjs/packageDetection-DT09yDE6.cjs.js
8.486 app:  - /app/node_modules/@backstage/cli/dist/cjs/buildBackend-FV4cZ7YT.cjs.js
8.486 app:  - /app/node_modules/@backstage/cli/dist/cjs/index-DtxQfUmH.cjs.js
8.486 app:  - /app/node_modules/@backstage/cli/dist/cjs/index-CS6xQRSZ.cjs.js
8.486 app:  - /app/node_modules/@backstage/cli/dist/index.cjs.js
8.486 app:  - /app/node_modules/@backstage/cli/bin/backstage-cli`


`npx @backstage/create-app --version   ->  1.27.0'
`yarn -v  ->  1.22.22`
`node -v  ->  v20.14.0`

--- a/packages/create-app/seed-yarn.lock
+++ b/packages/create-app/seed-yarn.lock
@@ -17,7 +17,7 @@
 // query: the version query to pin the version for, e.g. ^14.0.0
 // version: the version to pin to, must be in range of the query, e.g. 14.11.0
 
-"@swc/core@^1.3.46":
-  version "1.5.7"
-  resolved "https://registry.yarnpkg.com/@swc/core/-/core-1.5.7.tgz#e1db7b9887d5f34eb4a3256a738d0c5f1b018c33"
-  integrity sha512-U4qJRBefIJNJDRCCiVtkfa/hpiZ7w0R6kASea+/KLp+vkus3zcLSB8Ub8SvKgTIxjWpwsKcZlPf5nrv4ls46SQ==
+ajv@^8.0.0, ajv@^8.10.0, ajv@^8.11.0, ajv@^8.12.0, ajv@^8.14.0, ajv@^8.6.0, ajv@^8.6.3, ajv@^8.8.2, ajv@^8.9.0:
+  version "8.14.0"
+  resolved "https://registry.yarnpkg.com/ajv/-/ajv-8.14.0.tgz#f514ddfd4756abb200e1704414963620a625ebbb"
+  integrity sha512-oYs1UUtO97ZO2lJ4bwnWeQW8/zvOIQLGKcvPTsWmvc2SYgBb+upuNS5NxoLaMU4h8Ju3Nbj6Cq8mD2LQoqVKFA==

